src/discussions/learners/LearnerPostsView.test.jsx

```diff
@@ -62,31 +74,68 @@ describe('LearnerPostsView', () => {
 
     store = initializeStore();
     Factory.resetAll();
-    const learnerPosts = Factory.build('learnerPosts', {}, {
-      abuseFlaggedCount: 1,
-    });
-    const apiUrl = `${coursesApiUrl}${courseId}/learner/`;
     axiosMock = new MockAdapter(getAuthenticatedHttpClient());
-    axiosMock.onGet(apiUrl, { username, count_flagged: true })
-      .reply(() => [200, learnerPosts]);
+    axiosMock.onGet(learnerPostsApiUrl(courseId), { username, count_flagged: true })
+      .reply(() => [200, Factory.build('learnerPosts', {}, {
+        abuseFlaggedCount: 1,
+      })]);
+    await executeThunk(fetchUserPosts(courseId), store.dispatch, store.getState);
   });
 
-  describe('Basic', () => {
-    test('Reported icon is visible to moderator for post with reported comment', async () => {
-      axiosMock.onGet(`${courseConfigApiUrl}${courseId}/`).reply(200, {
-        has_moderation_privileges: true,
-      });
-      axiosMock.onGet(`${courseConfigApiUrl}${courseId}/settings`).reply(200, {});
-      await executeThunk(fetchCourseConfig(courseId), store.dispatch, store.getState);
-      await act(async () => {
-        renderComponent();
+  test('Reported icon is visible to moderator for post with reported comment', async () => {
+    await setUpPrivilages(axiosMock, store, true);
+    await waitFor(() => { renderComponent(); });
```

Comment:
you have use `await renderComponent();` in other test cases, please check if this test case will work the same as others

Reply:
There are situations where we have to wait until a loader disappears and a list appears. In such cases, it is necessary to use the "waitFor" function to ensure that the process completes before proceeding.